compiler/damlc/lib/DA/Cli/Damlc/Packaging.hs

```diff
@@ -71,22 +68,14 @@ import SdkVersion
 --   ledger. Based on the DAML-LF we generate dummy interface files
 --   and then remap references to those dummy packages to the original DAML-LF
 --   package id.
-createProjectPackageDb :: NormalizedFilePath -> Options -> PackageSdkVersion -> MS.Map UnitId GHC.ModuleName -> [FilePath] -> [FilePath] -> IO ()
-createProjectPackageDb projectRoot (disableScenarioService -> opts) (PackageSdkVersion thisSdkVer) modulePrefixes deps' dataDeps
+createProjectPackageDb :: NormalizedFilePath -> Options -> MS.Map UnitId GHC.ModuleName -> IO ()
+createProjectPackageDb projectRoot (disableScenarioService -> opts) modulePrefixes
   = do
-    deps <- expandSdkPackages (optDamlLfVersion opts) (filter (`notElem` basePackages) deps')
-    (needsReinitalization, depsFingerprint)
-        <- dbNeedsReinitialization projectRoot deps thisSdkVer (show $ optDamlLfVersion opts)
+    allDepFiles <- listFilesRecursive depsDir
+    (needsReinitalization, depsFingerprint) <- dbNeedsReinitialization projectRoot allDepFiles
```

Comment:
Can we just move listing files into `dbNeedsReinitialization`?

Reply:
moved it.